From 37b9994b9af7180ce0caf351779d0bdb73942160 Mon Sep 17 00:00:00 2001
From: David Brown <david.brown@linaro.org>
Date: Tue, 3 May 2016 14:00:17 -0600
Subject: [PATCH 2/2] Allow the kernel command line to be configurable

Pass the command line for the kernel through from the
BIOS_COMMAND_LINE make variable.  Also include BIOS_ROOT_DEVICE so
that the invocation of make can set this to a 'root=/dev/xxx' to
specify a root device.

Signed-off-by: David Brown <david.brown@linaro.org>
---
 bios/bios.mk | 1 +
 bios/conf.mk | 5 +++++
 bios/main.c  | 3 +--
 3 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/bios/bios.mk b/bios/bios.mk
index 5b4f999..3a021f9 100644
--- a/bios/bios.mk
+++ b/bios/bios.mk
@@ -2,6 +2,7 @@ include bios/conf.mk
 
 cppflags += -DPLATFORM_FLAVOR=PLATFORM_FLAVOR_ID_$(PLATFORM_FLAVOR)
 cppflags += -Iinclude
+cppflags += -DCOMMAND_LINE="\"$(BIOS_COMMAND_LINE)\""
 
 #
 # Do libraries
diff --git a/bios/conf.mk b/bios/conf.mk
index c7cc9a7..3d05be7 100644
--- a/bios/conf.mk
+++ b/bios/conf.mk
@@ -4,6 +4,11 @@ include mk/gcc.mk
 
 PLATFORM_FLAVOR ?= vexpress
 
+BIOS_COMMAND_LINE = console=ttyAMA0,115200 \
+		    earlyprintk=serial,ttyAMA0,115200 \
+		    dynamic_debug.verbose=1 \
+		    $(BIOS_ROOT_DEVICE)
+
 cpuarch = cortex-a15
 cflags	 = -mcpu=$(cpuarch) -mthumb
 cflags	+= -mthumb-interwork -mlong-calls
diff --git a/bios/main.c b/bios/main.c
index 46732d1..bce48fa 100644
--- a/bios/main.c
+++ b/bios/main.c
@@ -621,8 +621,7 @@ static void call_kernel(uint32_t entry, uint32_t dtb,
 {
 	kernel_ep_func ep = (kernel_ep_func)entry;
 	void *fdt = (void *)dtb;
-	const char cmdline[] =
-"console=ttyAMA0,115200 earlyprintk=serial,ttyAMA0,115200 dynamic_debug.verbose=1";
+	const char cmdline[] = COMMAND_LINE;
 	int r;
 	const uint32_t a0 = 0;
 	/*MACH_VEXPRESS see linux/arch/arm/tools/mach-types*/
-- 
2.8.2

